omniscidb/QueryEngine/Execute.cpp

```diff
@@ -1796,6 +1813,79 @@ hdk::ResultSetTable Executor::finishStreamExecution(
   return result;
 }
 
+std::pair<std::unique_ptr<policy::ExecutionPolicy>, ExecutorDeviceType>
```

Comment:
Don't we have the execution device types captured by the `ExecutionPolicy`?

Reply:
previously, the device type from `CompilationOptions` was used in `collectAllDeviceResults` and for non-heterogeneous execution. I left it to preserve what has already been written. It seemed to me that it was not necessary to store this type, because, indeed, everything is in `exe_policy`. but in order not to accidentally break anything, it seemed to me that it was better to leave the logic that is already written in `executeWorkUnitImpl`, so as not to mix refactoring of this section and the integration of the cost model in one pull request